File: app/src/main/java/com/konektedi/vs/Elections/ElectionView.java
package com.konektedi.vs.Elections;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.widget.Button;
import android.widget.GridView;
import android.widget.Toast;

import com.konektedi.vs.Elections.Results.ResultsView;
import com.konektedi.vs.R;

import java.util.ArrayList;
import java.util.List;

public class ElectionView extends AppCompatActivity {
    List<CategoriesModel> categoriesModelList;
    GridView categoriesGridView;
    Button resultsViewBtn, reviewsViewBtn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.election_view_activity);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        categoriesGridView = findViewById(R.id.categoriesGridView);
        resultsViewBtn = findViewById(R.id.resultsViewBtn);
        reviewsViewBtn = findViewById(R.id.reviewsViewBtn);

        getCategories();

        resultsViewBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(ElectionView.this, ResultsView.class);
                startActivity(intent);
            }
        });

        reviewsViewBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Toast.makeText(ElectionView.this, "Live Votes",
                        Toast.LENGTH_SHORT).show();
            }
        });

    }

    private void getCategories() {

        categoriesModelList = new ArrayList<>();

        for (int i = 0; i < 6; i++) {

            CategoriesModel category = new CategoriesModel();

            category.setCategory_id(i);
            category.setCategory("President category " + i);

            categoriesModelList.add(category);
        }

        categoriesGridView.setAdapter(new CategoriesAdapter(getApplication(), categoriesModelList));

    }


}

File: app/src/main/java/com/konektedi/vs/elections/ElectionView.kt
package com.konektedi.vs.elections

import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.LinearLayoutManager
import android.util.Log
import android.view.MenuItem
import com.konektedi.vs.R
import com.konektedi.vs.categories.CategoriesAdapter
import com.konektedi.vs.categories.CategoriesViewModel
import com.konektedi.vs.statistics.StatsAdapter
import com.konektedi.vs.utilities.common.Constants
import com.konektedi.vs.utilities.common.NetworkState
import kotlinx.android.synthetic.main.election_view_activity.*
import kotlinx.android.synthetic.main.election_view_content.*
import org.jetbrains.anko.design.longSnackbar
import android.graphics.PorterDuff
import androidx.core.content.ContextCompat
import com.konektedi.vs.R.id.toolbar
import android.graphics.drawable.Drawable



class ElectionView : AppCompatActivity() {
    private lateinit var viewModel: CategoriesViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.election_view_activity)
        setSupportActionBar(toolbar)

        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        val drawable = toolbar.navigationIcon
        drawable!!.setColorFilter(ContextCompat.getColor(this, R.color.accent_color), PorterDuff.Mode.SRC_ATOP)

        showCategories()
        swipeRefreshLayout.setOnRefreshListener { showCategories() }
    }

    private fun showCategories() {
        viewModel = ViewModelProviders.of(this).get(CategoriesViewModel::class.java)

        val data = intent.extras

        var electionId = 0
        var electionTitle = ""

        if (data != null) {
            electionId = data.getInt(Constants.ELECTION_ID)
            electionTitle = data.getString(Constants.ELECTION_TITLE)
        }

        title = electionTitle

        viewModel.getCategories(electionId).observe(this, Observer {
            recyclerView.layoutManager = androidx.recyclerview.widget.LinearLayoutManager(this@ElectionView)
            val adapter = CategoriesAdapter(this@ElectionView, it!!)
            recyclerView.adapter = adapter
        })

        viewModel.getStats(electionId).observe(this, Observer {
            val columns = 2
            statsRecyclerView.layoutManager = androidx.recyclerview.widget.GridLayoutManager(this@ElectionView, columns)
            val adapter = StatsAdapter(this@ElectionView, it!!)
            statsRecyclerView.adapter = adapter
        })

        viewModel.networkState.observe(this, Observer {
            when (it) {
                NetworkState.LOADING -> showProgress(true)
                NetworkState.LOADED -> showProgress(false)
                else -> {
                    showProgress(false)
                    coordinatorLayout.longSnackbar(it?.msg.toString())
                }
            }
        })
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> super.onBackPressed()
        }
        return true
    }

    private fun showProgress(boolean: Boolean) {
        swipeRefreshLayout.isRefreshing = boolean
    }
}
